lib/Image/Adapter/Imagick.php

```diff
@@ -752,7 +752,7 @@ public function addOverlay(mixed $image, int $x = 0, int $y = 0, int $alpha = 10
             }
 
             $newImage->evaluateImage(\Imagick::EVALUATE_MULTIPLY, $alpha, \Imagick::CHANNEL_ALPHA);
-            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), $x, $y);
+            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), intval($x), intval($y));
```

Comment:
I think we should just cast:
```suggestion
            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), (int)$x, (int)$y);
```